### ISSUE TITLE
Add network diagnostic mode to Scylla

### ISSUE THREAD

**Issue by dorlaor** [scylladb/scylla#4048]
Some times network outages cause issues which are hard to trace. Network issues
may come and go, some may stay for a long while and the opposite.

It's desirable to have a network diagnostic in scylla so once we configure (dynamically, maybe through restapi or virtual table) a running node to enter diagnostic mode that will do:
 - Open and close x number of connections with dummy data on them
 - All existing ports should be part of it - extend the rpc api with a test verb
 - Report the success of the above
 - If needed, use encryption

**Comment by amnonh** [comment]
Unless we want the node to behave differently during diagnostic (in the sense that it will handle a lower load, or temporarily put itself outside of the cluster) why do it inside scylla?

It's very much like the CQL ping functionality @mmatczuk implemented in the manager, isn't it safer and easier to create such a functionality completely standalone (plus it can work even if scylla nodes are crushing at the time)

BTW even if you do want the node to let the cluster node it wants a lower load, we can limit only that functionality to scylla and do the rest in an external app

**Comment by vladzcloudius** [comment]
@dorlaor I agree with @amnonh that we'd rather have this as a stand alone tool for case when `scylla` is unable to start (e.g. due to networking issues).

**Comment by dorlaor** [comment]
Agree that scylla manager's cql health is a good idea but we need more than
a single idea.
That specific customer issue was about the rpc port (7000), so it could be
any node (not a cluster issue) or
any scylla service. There is no silver bullet and we need couple of
mechanisms .

**Comment by gleb-cloudius** [comment]
Part of the gossiper functionality is to check for network outages and
for most simple cases this is usually enough. I do not know what
customer issue was that motivated you to open this issue, but in the
past the problematic scenarios we encountered were some weird NATs that
forgot some long running connection (not even sending reset on them,
just starting to drop packets) and nothing from the functionality you
proposed will solve that.

--
			Gleb.

**Comment by mmatczuk** [comment]
My 2c
if feels like we could have a bunch on nmap scripts to run as a smoke test to automate troubleshooting process but I think its usability is limited.

**Comment by dorlaor** [comment]
@gleb-cloudius this is another customer, not with nat issue but with intermediate AWS issue.
There is no silver bullet but it will be nice that Scylla core can run various validity checks, network is one of them it can be:
  - network tests:
     - send/receive 10 packets with {one node, all local nodes, all nodes within a DC, all nodes)
     - With/without encryption
     - all ports
  - Disk tests
     - Check all major directories: commitlog, data,..
  - Anything else

Scylla manager can trigger those tests from a cluster perspective and collect the results.
